packages/webapp/src/components/Filter/FilterDate/index.jsx

```diff
@@ -21,9 +21,9 @@ export function FilterDate({ defaultValue, onChange, subject, shouldReset, props
   const [date, setDate] = useState(defaultValue ?? '');
 
   useEffect(() => {
-    if (shouldReset) {
+    if (shouldReset && date) {
       setDate('');
-      onChange('');
+      onChange(undefined);
```

Comment:
I changed this to `undefined` from empty string as an empty string will show the filter icon with no value
![image](https://github.com/user-attachments/assets/ed221f68-4cdf-4357-a9a8-0427d3a9937d)